### DISCUSSION TITLE
Recommended for Mac... Docker compose or other Mac  install? Confused

### DISCUSSION THREAD

**Discussion by ewebgh33** [danny-avila/LibreChat#1947] [General]
Because in the docs when you go to Mac Install, in big letters up the top it says Docker Compose recommended.
Then it handily says they have videos for Windows and Linux... but not Mac.

Then I go to find out about Docker on Mac, go to the Docker docs, 
which says Docker Compose is only for Win and Linux. There's even a big orange warning box.
https://docs.docker.com/compose/install/

So if this is the case, why is the recommended install Docker Compose, when I go to the Mac Install page for Libre docs?

And bonus question, we still have to set up Mongo or do you have some alt embedding library methods available yet?
Thanks
E

**Comment by fuegovic** [comment]
That big orange warning in the link you shared is only about the fact that compose v1 has been deprecated, and that we should use compose v2.
Docker install is recommended, and comes with mongodb (and meilisearch) included, no need to set it up yourself.

---

![image](https://github.com/danny-avila/LibreChat/assets/32828263/c9d87c22-2c3c-48fa-b6aa-2802a0492023)

The basic steps should go something like this:
- install docker desktop: https://docs.docker.com/desktop/install/mac-install/
- clone the LibreChat repo: `git clone https://github.com/danny-avila/LibreChat.git`
- use the terminal to navigate to the root of the LibreChat folder
- create your `.env` file by duplicating the `.env.example` file: `cp .env.example .env`
- run this command: `docker compose up -d`
-  and that's pretty much it, when it's ready you should be able to access librechat at http://localhost:3080/

---
Start by getting it to run with the default settings, then you can start customizing the more advanced parameters in the .env or add custom endpoints with the `librechat.yaml` config file. Restart the docker container to apply the changes. Our docs have more info about advanced configuration: https://docs.librechat.ai/install/index.html

**Reply by ewebgh33** [reply]
Thanks very much, I appreciate the help here. That orange box isn't very clear about what it's referring to. I was also wondering if it was a flag for the text above or below the box.

Google results were a bit hacky for installing docker compose but that makes more sense if all I need is Desktop, and Compose v1 is old/deprecated.

Cheers
E
